api/v1alpha1/awsvalidator_types.go

```diff
@@ -40,6 +40,19 @@ type AwsAuth struct {
 	SecretName string `json:"secretName,omitempty" yaml:"secretName,omitempty"`
 	// Option 2: specify a service account (EKS)
 	ServiceAccountName string `json:"serviceAccountName,omitempty" yaml:"serviceAccountName,omitempty"`
+	// STS authentication properties (optional, to be provided in conjunction with Option 1 or 2)
+	StsAuth AwsSTSAuth `json:"stsAuth,omitempty" yaml:"stsAuth,omitempty"`
```

Comment:
Please make the `StsAuth` field an `*AwsSTSAuth`.

That way you can do the following in `aws.go`:
```golang
if validator.Spec.Auth.StsAuth != nil {
	awsStsConfig(&cfg, validator.Spec.Auth.StsAuth)
}
```
Otherwise, `&validator.Spec.Auth.StsAuth` would never be `nil`.